flag_groups.go

```diff
@@ -157,3 +157,51 @@ func sortedKeys(m map[string]map[string]bool) []string {
 	sort.Strings(keys)
 	return keys
 }
+
+// enforceFlagGroupsForCompletion will do the following:
+// - when a flag in a group is present, other flags in the group will be marked required
+// - when a flag in a mutually exclusive group is present, other flags in the group will be marked as hidden
+// This allows the standard completion logic to behave appropriately for flag groups
+func (c *Command) enforceFlagGroupsForCompletion() {
+	if c.DisableFlagParsing {
+		return
+	}
+
+	groupStatus := map[string]map[string]bool{}
+	mutuallyExclusiveGroupStatus := map[string]map[string]bool{}
+	c.Flags().VisitAll(func(pflag *flag.Flag) {
+		processFlagForGroupAnnotation(pflag, requiredAsGroup, groupStatus)
```

Comment:
So the way I just modified the other PR was that I pass the flagset into this method and the resulting groupStatus map only has the flag groups of interest.

So by incorporating the new code you may avoid the other problem areas I noted.

Reply:
Ha!  I should have read all your comments first before replying 😄 
Ok then, no need to add the nil check